contracts/near/src/lib.rs

```diff
@@ -0,0 +1,437 @@
+        self.source.remove(&source);
+    }
+
+    pub fn get_sources(&self) -> Vec<Source> {
+        self.source.iter().collect()
+    }
+
+    /// Process a Batch of Price Attestations.
+    ///
+    /// This receives price updates, verifies them, and stores the attestations in the contract
+    /// state allowing other contracts to query them.
+    #[private]
+    pub fn attest_prices(&mut self, batch: &BatchPriceAttestation) -> usize {
+        let mut count_updates = 0;
+        for price_attestation in &batch.price_attestations {
+            // Verify the PriceAttestation should be accepted, and if so, store it.
+            if update_price_feed_if_new(&mut self.prices, PriceFeed::from(price_attestation)) {
+                count_updates += 1;
+            }
+        }
+        count_updates
+    }
+
+    /// Read the current Price for a given Price Feed, if the price is past the staleness threshold
+    /// the trading status is set to Unknown before returning.
+    pub fn get_price(&self, price_id: Identifier) -> Option<PriceFeed> {
+        self.prices.get(&price_id).and_then(|mut price| {
+            let block_timestamp = env::block_timestamp();
+            let price_timestamp = price.publish_time.try_into().ok()?;
+
+            // - If Price older than STALENESS_THRESHOLD, set status to Unknown.
+            // - If Price newer than now by more than STALENESS_THRESHOLD, set status to Unknown.
+            // - Any other price around the current time is considered valid.
+            if u64::abs_diff(block_timestamp, price_timestamp) > STALENESS_THRESHOLD {
+                price.status = PriceStatus::Unknown;
+            }
+
+            Some(price)
+        })
+    }
+
+    /// Instruction for processing VAA's relayed via Wormhole.
+    ///
+    /// Note that VAA verification requires calling Wormhole so processing of the VAA itself is
+    /// done in a callback handler, see `process_vaa_callback`.
+    #[payable]
+    #[handle_result]
+    pub fn process_vaa(&mut self, vaa: String) -> Result<(), Error> {
+        // Verify the VAA is coming from a trusted source chain.
+        {
+            let vaa = hex::decode(&vaa).map_err(|_| Error::InvalidHex)?;
+            let vaa = vaa_payload::from_slice_with_payload::<structs::Vaa<()>>(&vaa)
+                .map_err(|_| Error::InvalidVaa)?;
+
+            if !self.source.contains(&Source {
+                emitter:            vaa.0.emitter_address.0,
+                pyth_emitter_chain: vaa.0.emitter_chain as u16,
+            }) {
+                return Err(Error::UnknownSource);
+            }
+        }
+
+        // Verify VAA and refund the caller in case of failure.
+        ext::ext_worm_hole::ext(env::current_account_id())
+            .with_static_gas(Gas(30_000_000_000_000))
+            .verify_vaa(vaa.clone())
+            .then(
+                Self::ext(env::current_account_id())
+                    .with_static_gas(Gas(30_000_000_000_000))
```

Comment:
I recall that Josh was manually checking gas on each call and did not remember this syntax, is this new to ensure fixed remaining gas?